src/scheduler/workers/analytics/addAnalyticsAuditEvent.ts

```diff
@@ -0,0 +1,40 @@
+import { prisma } from "scheduler/workers/model";
+
+const truncateMessage = "... truncated due to max length";
+const MAX_BYTES = 7000 - Buffer.byteLength(truncateMessage);
+
+export async function addAnalyticsAuditEvent(view: string, dataSent: any[], response: any) {
+  const maxSize = MAX_BYTES - getByteLength(JSON.stringify({ data: "", response }));
+  const truncatedData = truncateString(JSON.stringify(dataSent), maxSize);
+
+  await prisma.audit.create({
+    data: {
+      auditEvent: "ANALYTICS",
+      createdAt: new Date(),
+      jsonData: {
+        view,
+        dataSent: truncatedData,
+        response,
+      },
+    },
+  });
+}
+
+function getByteLength(string: string) {
+  return Buffer.byteLength(string);
+}
+
+function truncateString(string: string, maxByteLength: number, hasTruncated = false): string {
+  /**
+   * Recursive function which will keep removing characters until the string is the byte under limit.
+   * JS (node 14) does not natively support truncating/slicing by byte length, and due to UTF-16,
+   * characters may be 2-5+ bytes long, hence "estimatedMaxCharLength"
+   */
```

Comment:
Is there a version of node that does support it? If so it would be good to know in case we upgrade the project so we can refactor this code

Reply:
I believe 16 with the `Blob` object, so I'll add that